### PR TITLE
Add Azure Function for GetUploadStatusById

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Adding Azure Function for GetUploadStatusById

## Why?

Completing subtasks NAC-918 & NAC-916 for story in NAC-59

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: etl/tests/Piipan.Etl.Func.BulkUpload.Tests/GetUploadStatusApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            var function = new GetUploadStatusApi(participantUploadApi.Object);
+
+            // Act
+            var result = await function.GetUploadById(request, "upload1", logger.Object) as JsonResult;
+            var response = result.Value as UploadStatusApiResponse;
+
+            // Assert
+            Assert.Same(uploadDto, response.Data);
+
+            //VerifyLogError(logger, "No input stream was provided");
```
Comment (dborgen2) [inline etl/tests/Piipan.Etl.Func.BulkUpload.Tests/GetUploadStatusApiTests.cs:41]: Nitpick: Looks like we had some logic here that you commented out. I'm assuming we can remove it?

**Review by dborgen2 — APPROVED** [review]
LGTM, just a couple small non-blocking questions

File: etl/tests/Piipan.Etl.Func.BulkUpload.Tests/GetUploadStatusApiTests.cs (hunk 1 above)
Comment (bgfoshay) [reply etl/tests/Piipan.Etl.Func.BulkUpload.Tests/GetUploadStatusApiTests.cs:41]: Yep, you're right. I'll get rid of it.

File: etl/src/Piipan.Etl/Piipan.Etl.Func.BulkUpload/Models/UploadStatusApiResponse.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Piipan.Etl.Func.BulkUpload.Models
+{
+    /// <summary>
+    /// Azure Function GetUploadStatus Response
+    /// </summary>
+    public class UploadStatusApiResponse
+    {
+        [JsonProperty("data", Required = Required.Always)]
+        public UploadDto Data { get; set; }
```
Comment (edwintorres) [inline etl/src/Piipan.Etl/Piipan.Etl.Func.BulkUpload/Models/UploadStatusApiResponse.cs:12]: Non-block question: the name Data suggest this is going to be re-use for different scenarios. Am I correct?

**Review by edwintorres — APPROVED** [review]

File: participants/src/Piipan.Participants/Piipan.Participants.Core/DataAccessObjects/UploadDao.cs [hunk 3]
```diff
@@ -80,6 +80,9 @@ FROM uploads
         /// <summary>
         /// Retrieves upload record by id
         /// </summary>
+        /// <remarks>
+        /// Throws InvalidOperationException if 0 or more than 1 record is found.
```
[the remark this answers is not in the document]
Comment (bgfoshay) [reply participants/src/Piipan.Participants/Piipan.Participants.Core/DataAccessObjects/UploadDao.cs:84]: Good question. The query actually ensures that only one record can ever be returned. I'll update the documentation.

**Pull request merged**